### PR TITLE
Fix memory leaks

### LINKED ISSUE #17030: Memory leak in [platform/android/export/export.cpp:1523]
**Godot version:**

79a225ac2a2f2353f2f626809b56229397dfe52d

**OS/device including version:**

**Issue description:**

We find a memory leak defect in the file  tools/winebuild/po.c.
I've uploaded a screenshot of the leak trace.

Gang
Sourcebrella

**Steps to reproduce:**
screenshot from 2018-02-26 16 41 38
screenshot from 2018-02-26 16 42 05
screenshot from 2018-02-26 16 42 16

**Minimal reproduction project:**

### LINKED ISSUE #17033: Memory Leak in [mkvparser.cc:1283]
**Godot version:**

79a225ac2a2f2353f2f626809b56229397dfe52d

**OS/device including version:**

**Issue description:**

pCluster leaks when this function returns on line 1289 or  1291

**Steps to reproduce:**
screenshot from 2018-02-26 16 49 15

**Minimal reproduction project:**

### PR DESCRIPTION
**Pull request by Chaosus**
Closes #17033, #17032, #17031, thanks @fangang190

Im not sure about right way to fix for #17030, cuz im not an expert in android apk, and calling zipClose(unaligned_apk, NULL); before return may cause errors, so anybody with free time and knowledge could fix it instead

### CODE REVIEW

**Comment by Noshyaar** [comment]
Thirdparty bugfix should contain comment `// -- GODOT start --` and `// -- GODOT end --`, a .patch of the change, and the following comment to thirdparty/README.md:

```
Important: Some files have Godot-made changes. They are marked with // -- GODOT start -- 
and // -- GODOT end -- comments and a patch is provided in the [thirdparty_module]/ folder.
```

**Comment by Chaosus** [comment]
@Noshyaar I remember this, thanks, I fixed it, its ok now ?

**Comment by Chaosus** [comment]
Damn, I didnt notice that actual thirdparty/README.md is existed.. Sorry, I will fix this..

**Comment by Chaosus** [comment]
Done.. waiting for merge..

**Comment by Hinsbart** [comment]
Did you forget to `git add` the patches by any chance? They seem to be missing ^^

**Comment by Chaosus** [comment]
I dont understand you, I just use the commit and then rebase with squash.. and it was enough in most cases

**Comment by Hinsbart** [comment]
Well, what I mean is that you mention the patches in the readme:
> Important: Some files have Godot-made changes.
They are marked with `// -- GODOT start --` and `// -- GODOT end --`
comments and a patch is provided in the b2d_convexdecomp/ folder.

But there are no *.patch files in your commit.

**Comment by Chaosus** [comment]
@Hinsbart Ok, I change it to just

"Important: Some files have Godot-made changes.
They are marked with `// -- GODOT start --` and `// -- GODOT end --`
comments."

File: thirdparty/libsimplewebm/libwebm/mkvparser/mkvparser.cc [hunk 1]
```diff
@@ -1284,7 +1284,9 @@ long Segment::DoLoadCluster(long long& pos, long& len) {
     pos += cluster_size;
 
     m_pos = pos;
-
+    // -- GODOT start --
+    delete pCluster;
```
Comment (akien-mga) [inline thirdparty/libsimplewebm/libwebm/mkvparser/mkvparser.cc]: Won't `pCluster` leak on line 1297 too if we don't enter this `if`?

Comment (akien-mga) [reply thirdparty/libsimplewebm/libwebm/mkvparser/mkvparser.cc]: Nevermind, it's assigned to `m_pUnknownSize` and should not be freed.

Comment (Chaosus) [reply thirdparty/libsimplewebm/libwebm/mkvparser/mkvparser.cc]: Yeah, I saw it too and dont decide to delete

**Comment by akien-mga** [comment]
Thanks! Would you mind contributing those fixes upstream too, so that we can drop our custom modification next time we sync with them?
- b2d_convexdecomp: https://github.com/erincatto/Box2D/blob/master/Contributions/Utilities/ConvexDecomposition/b2Polygon.cpp#L1340
- Etc2comp: https://github.com/google/etc2comp/blob/master/EtcLib/Etc/EtcFilter.cpp#L229-L232
- libwebm: That's Google, no clue how to send a PR, so maybe leave that one..

**Comment by Chaosus** [comment]
Okay ) I take a look on it

**Comment by Chaosus** [comment]
@akien-mga Seems like erincato merged my PR to b2Polygon - https://github.com/erincatto/Box2D/pull/486,
I think the part of this PR should be reverted and b2Polygon should be updated

**Comment by akien-mga** [comment]
Thanks for the notice. It looks like we already had quite a few changes compared to the upstream code (diff when trying to replace our files by upstream ones: https://paste.fedoraproject.org/paste/-5ujnPQIUTZFZI8IjhhB2A), so I guess it's good to keep as is.

Some of our changes shown in the diff are cosmetic and could be pushed upstream, but others are specific to Godot (like removing the dependency on Box2D's glue and using some Godot constants like `CMP_EPSILON`, or adding a namespace and removing debug spam).

**Pull request merged**